app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -277,10 +277,11 @@ private void doMenuAction(int menuItemId) {
                 smoothActionBarToggle.runWhenIdle(new Runnable() {
                     @Override
                     public void run() {
-                        startActivity(intent);
                         overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_left);
                     }
                 });
+                startActivity(intent);
```

Comment:
the reason is that the settings activity is a different activity altogether, its not a fragment, the rest of the options are all fragment. No need to show me a gif, I know about it. when you call activity#onCreate the previous must be destroyed(or paused) before switching. And since the drawer is attached to it therefore we need to wait for the animation to finish. 

Yes break statement is needed, that's all. I don't think you need to do anything more than that.